### PR TITLE
fix the failing lint tests

### LINKED ISSUE #506: fix the golang-ci lint errors
#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v
- **Meshery Adapter Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 📝 [Meshery Adapters Spreadsheet](https://docs.google.com/spreadsheets/d/1rGGpSXC68iDJzNRU-qZIRQsZUwkt1qLFg7JCs4pfvbU/edit?usp=sharing)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ayushthe1**
**Description**

This PR fixes #506 

**Notes for Reviewers**
Some tests are still failing. To fix those ,some changes would have to be done in the `meshery-adapter-library` .

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: linkerd/linkerd.go [hunk 1]
```diff
@@ -224,8 +225,8 @@ func (linkerd *Linkerd) ProcessOAM(ctx context.Context, oamReq adapter.OAMReques
 	kubeconfigs := oamReq.K8sConfigs
 	var comps []v1alpha1.Component
 	for _, acomp := range oamReq.OamComps {
-		comp, err := oam.ParseApplicationComponent(acomp)
-		if err != nil {
+		comp, parseErr := oam.ParseApplicationComponent(acomp)
```
Comment (Aisuko) [inline linkerd/linkerd.go:228]: Meshery stop support OAM already. We can ignore OAM-related functions.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Please check the CI result and update the code. Your work is so clear.

**Comment by Aisuko** [comment]
Hi @ayushthe1 Thanks for your contribution, there are still some errors from the CI. Could you solve all of them? Maybe some mistakes should be ignored.

**Comment by ayushthe1** [comment]
@Aisuko Could you specify which lint tests should i ignore ?Also could you please review the changes i have done to the `init` functions .I have some doubt on them.

**Comment by Aisuko** [comment]
> @Aisuko Could you specify which lint tests should i ignore ?Also could you please review the changes i have done to the `init` functions .I have some doubt on them.

Hi @ayushthe1, 

Here is result from `golangci-lint`, if you can help to fix this, it would be good, but I know it is not easy to solve. So, if it can cause any breaking change, please ignore them in this PR,

For Golang code.

```
run golangci-lint
  Running [/home/runner/golangci-lint-1.52.2-linux-amd64/golangci-lint run --out-format=github-actions --timeout 3m --verbose] in [] ...
  Error: hugeParam: opReq is heavy (112 bytes); consider passing it by pointer (gocritic)
  Error: hugeParam: oamReq is heavy (88 bytes); consider passing it by pointer (gocritic)
  Error: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
  Error: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
  Error: named return "t1" with type "time.Time" found (nonamedreturns)
  Error: named return "loc" with type "string" found (nonamedreturns)
```

For E2E Tests part, we can ignore this part currently. We are working to make sure the E2E test can be useful in the future.

File: internal/config/releases.go [hunk 2]
```diff
@@ -70,7 +70,14 @@ func getLatestReleases(releases uint) ([]*Release, error) {
 	releaseAPIURL := "https://api.github.com/repos/linkerd/linkerd2/releases?per_page=" + fmt.Sprint(releases)
 	// We need a variable url here hence using nosec
 	// #nosec
-	resp, err := http.Get(releaseAPIURL)
+	client := &http.Client{}
+
+	req, err := http.NewRequest("GET", releaseAPIURL, http.NoBody)
```
Comment (Aisuko) [inline internal/config/releases.go:75]: Is there I have considered the suggestion from `golangci-lint`?

`should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)`

File: linkerd/cert/cert.go [hunk 3]
```diff
@@ -93,6 +93,6 @@ func GetDefaultX509Cert(serialNumber uint64, k *ecdsa.PublicKey, validFrom *time
 }
 
 // GetWindow returns cert validity window based on the arguments given
-func GetWindow(t time.Time, lifetime, clockSkewAllowance time.Duration) (time.Time, time.Time) {
+func GetWindow(t time.Time, lifetime, clockSkewAllowance time.Duration) (t1, t2 time.Time) {
```
Comment (Aisuko) [inline linkerd/cert/cert.go:96]: Please check the CI feedback here.  `named return "t1" with type "time.Time" found (nonamedreturns)`

File: linkerd/install.go [hunk 4]
```diff
@@ -209,7 +209,7 @@ func (linkerd *Linkerd) applyHelmChart(appversion string, namespace string, isDe
 	return nil
 }
 
-func getChartLocationAndVersion(version string) (string, string) {
+func getChartLocationAndVersion(version string) (loc, ver string) {
```
Comment (Aisuko) [inline linkerd/install.go:212]: Same here.

File: linkerd/install.go [hunk 5]
```diff
@@ -336,15 +339,17 @@ func (linkerd *Linkerd) getExecutable(release string) (string, error) {
...
-	case "windows":
+	case "darwin", "windows":
 		url = fmt.Sprintf("%s/%s/linkerd2-cli-%s-%s", url, release, release, platform)
 	case "linux":
 		url = fmt.Sprintf("%s/%s/linkerd2-cli-%s-%s-%s", url, release, release, platform, arch)
 	}
-
-	resp, err := http.Get(url)
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, http.NoBody)
```
Comment (Aisuko) [inline linkerd/install.go:348]: Same here.

**Review by Aisuko — CHANGES_REQUESTED** [review]

**Comment by ayushthe1** [comment]
Thanks for reviewing @Aisuko .I have solved all the error except one. There is a test called `nonamedresults` which i have disabled because this and one another test (gocritic) gives conflicting suggestions .Like if you solve the `nonamedresults`, then the other error will pop out and vice-versa. 
The one error that is still left to be resolved will require changes in the `Adapter` struct inside the `meshery-adapter-library`. Should i go forward with making those changes ? Also can you please take a look at changes i have done to the `init()` function in this PR where i have replaced them.

**Pull request closed without merging**